Migrate Axes Display

Martin-Idel opened the pull request:
Closes #93 

- Migrate display and fix linters
- Add visual test for axes display.
- Do not display axes when transformation fails
- Small refactoring of axes class for c++14

Martin-Idel committed:
Move axes display

Martin-Idel committed:
Make display compile and fix linters

Martin-Idel committed:
Hide axis when transform in error

Martin-Idel committed:
Minor refactoring of axes

- Shapes are now unique pointers
- When accessing them, return a reference only (no need for anything
else

Martin-Idel committed:
Add visual test

Martin-Idel committed:
Update README

jacobperron commented on rviz_default_plugins/plugins_description.xml:
```diff
@@ -2,6 +2,17 @@
   <!-- Displays plugins -->
 
+  <class
+          name="rviz_default_plugins/Axes"
```
nit: the attributes look over-indented

jacobperron reviewed (commented):
Sorry for taking so long to get to this.

LGTM :+1: After a rebase I'll trigger CI.

Martin-Idel committed:
Fix indentation in plugins_description

Martin-Idel commented:
Rebase done.

I also took the liberty to tweak the readme: The "Oculus" display isn't part of this repository and the Interactive Marker display as well as message filters have been ported. On the other hand, stereo functionality is currently not enabled, so that should be put on the list. 

I can revert those changes if you want to, of course.

jacobperron commented:
I noticed that with the default value, `FIXED_FRAME`, it is unable to resolve the transformation even if the global fixed frame is available. I guess this is an issue to be resolved in the frame transformer. 
We can ticket it separately if you agree.

jacobperron commented:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=8245)](http://ci.ros2.org/job/ci_linux/8245/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=4212)](http://ci.ros2.org/job/ci_linux-aarch64/4212/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=6727)](http://ci.ros2.org/job/ci_osx/6727/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=8135)](http://ci.ros2.org/job/ci_windows/8135/)

Edit: re-triggered with dbe74f6

jacobperron commented on rviz_default_plugins/src/rviz_default_plugins/displays/axes/axes_display.cpp:
```diff
@@ -0,0 +1,129 @@
+  (void) dt;
+  (void) ros_dt;
+
+  std::string frame = frame_property_->getStdString();
```
Here's the bug, it should be `getFrameStd()` instead:
```suggestion
  std::string frame = frame_property_->getFrameStd();
```

jacobperron replied:
See dbe74f6

jacobperron committed:
Use getFrameStd instead of getStdString

To ensure we get a frame ID in the case the property is set to the fixed frame.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

Martin-Idel commented:
Thanks for catching this. I only retested with the visual test and some frame transformations. Looks good to me.
